src/renderers/dom/shared/eventPlugins/ChangeEventPlugin.js

```diff
@@ -289,6 +289,18 @@ function getTargetInstForInputOrChangeEvent(
   }
 }
 
+function handleControlledInputBlur(inst, node) {
+  // Fiber and ReactDOM keep wrapper state in separate places
+  let state = inst._wrapperState || node._wrapperState;
+
+  if (!state || !state.controlled || node.type !== 'number') {
+    return;
+  }
+
+  // If controlled, assign the value attribute to the current value on blur
+  node.setAttribute('value', '' + node.value);
```

Comment:
Hmm. Yes. I'll through a conditional around this that does a comparison of the value property. 

Could the input value tracker help us out here to avoid DOM access? cc @jquense